### PR TITLE
ICS10 `ClientReader` trait implementation for `Context`

### PR DESCRIPTION
**Pull request by vmarkushin**

### CODE REVIEW

**Comment by seunlanlege** [comment]
oh wow, i don't think we need the full `Reader` context implmentation, Only the the method for fetching `consensus_state`

File: light-clients/ics10-grandpa-cw/src/channel.rs [hunk 1]
```diff
@@ -1,195 +1,250 @@
...
+	}
+
+	fn connection_channels(
+		&self,
+		conn_id: &ConnectionId,
+	) -> Result<Vec<(PortId, ChannelId)>, Error> {
+		let conn_id_bytes = conn_id.as_bytes().to_vec();
+		let port_and_channel_id =
+			CHANNELS_CONNECTION.load(self.storage(), conn_id_bytes).unwrap_or_default();
+		if port_and_channel_id.is_empty() {
```
Comment (Wizdave97) [inline light-clients/ics10-grandpa-cw/src/channel.rs:58]: It should not return an error if it's empty, should just return an empty list.

File: light-clients/ics10-grandpa-cw/src/client.rs [hunk 2]
```diff
@@ -27,60 +33,137 @@ impl<'a, H: HostFunctions> ClientTypes for Context<'a, H> {
...
+			self,
+			"in client : [consensus_state] >> any consensus state = {:?}",
+			any_consensus_state
+		);
+		Ok(any_consensus_state)
 	}
 
 	fn host_client_type(&self) -> String {
-		todo!()
+		unimplemented!()
```
Comment (Wizdave97) [inline light-clients/ics10-grandpa-cw/src/client.rs:103]: Host type should be tendermint

**Comment by vmarkushin** [comment]
@seunlanlege ideally yes, because these methods can be used by ibc-go in other ICSs (right?) and by the client itself.

File: ibc/modules/src/core/ics23_commitment/commitment.rs [hunk 3]
```diff
@@ -21,7 +21,7 @@ use subtle_encoding::{Encoding, Hex};
 
 use super::merkle::MerkleProof;
 
-#[derive(Clone, PartialEq, Eq, Serialize)]
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize)]
 #[serde(transparent)]
 pub struct CommitmentRoot {
 	#[serde(serialize_with = "crate::serializers::ser_hex_upper")]
```
Comment (Wizdave97) [inline ibc/modules/src/core/ics23_commitment/commitment.rs:27]: You need the hex deserializer here, if you are deserializing it.

File: light-clients/ics10-grandpa-cw/src/lib.rs [hunk 4]
```diff
@@ -6,13 +6,14 @@ pub mod contract;
...
+mod macros;
 pub mod msg;
 pub mod state;
 mod types;
 
 pub use crate::error::ContractError;
 
 pub const CLIENT_STATE: &'static [u8] = b"client_state";
-pub const STORAGE_PREFIX: &'static [u8] = b"ibc";
+pub const CONNECTION_PREFIX: &'static [u8] = b"ibc";
```
Comment (Wizdave97) [inline light-clients/ics10-grandpa-cw/src/lib.rs:17]:
```suggestion
pub const CONNECTION_PREFIX: &'static [u8] = b"ibc/";
```

**Review by Wizdave97 — APPROVED** [review]
Looks good, save a few comments.

**Pull request merged**